cli/command/container/stop.go

```diff
@@ -39,16 +40,45 @@ func NewStopCommand(dockerCli *command.DockerCli) *cobra.Command {
 
 func runStop(dockerCli *command.DockerCli, opts *stopOptions) error {
 	ctx := context.Background()
+	timeout := time.Duration(opts.time) * time.Second
 
+	var waitAll sync.WaitGroup
 	var errs []string
+	const maxParallel int32 = 50
+
+	// make sure result is printed in order
+	output := map[string]chan error{}
 	for _, container := range opts.containers {
-		timeout := time.Duration(opts.time) * time.Second
-		if err := dockerCli.Client().ContainerStop(ctx, container, &timeout); err != nil {
-			errs = append(errs, err.Error())
-		} else {
-			fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+		output[container] = make(chan error, 1)
+	}
+	waitAll.Add(1)
+	go func() {
+		for _, container := range opts.containers {
+			if err := <-output[container]; err != nil {
+				errs = append(errs, err.Error())
+			} else {
+				fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+			}
 		}
+		waitAll.Done()
+	}()
+
+	var sem = make(chan struct{}, maxParallel)
+	waitAll.Add(len(opts.containers))
+	for _, container := range opts.containers {
+		sem <- struct{}{} // Wait for active queue sem to drain.
+		go func(container string) {
+			defer waitAll.Done()
+
+			if err := dockerCli.Client().ContainerStop(ctx, container, &timeout); err != nil {
+				output[container] <- err
+			} else {
+				output[container] <- nil
+			}
```

Comment:
No reason to check `err != nil` here. You're also doing the same action in both branches, just send directly to the channel:

```go
err := ...
output[container] <- err
```